workflow/scripts/get-transcript-info.R

```diff
@@ -39,69 +40,124 @@ while ( class(mart)[[1]] != "Mart" ) {
-        ) %>%
-        mutate_at(
-          vars(canonical),
-          function(values) {
-            as_vector(
-              map(
-                str_trim(values), 
-                function(v) {
-                  if (is.na(v)) {
-                    NA
-                  } else if (v == "1") { 
-                    TRUE 
-                  } else if (v == "0") {
-                    FALSE
-                  }
-                }
-              )
-            )
-          }
+  attributes <- c(attributes, "transcript_is_canonical",
+    "chromosome_name", "transcript_mane_select",
+      "ensembl_transcript_id_version")
+  t2g <- biomaRt::getBM(
+  attributes = attributes,
+  mart = mart,
+  useCache = FALSE
+  )
+  t2g <- t2g %>%
+    rename(
+      target_id = ensembl_transcript_id,
+      ens_gene = ensembl_gene_id,
+      ext_gene = external_gene_name,
+      gene_desc = description,
+      canonical = transcript_is_canonical,
+      chromosome_name = chromosome_name,
+      transcript_mane_select = transcript_mane_select,
+    ) %>%
+    filter(!str_detect(chromosome_name, "patch|PATCH")) %>%
+    filter(str_detect(transcript_mane_select, "")) %>%
+    mutate_at(
+      vars(gene_desc),
+      function(values) {
+        str_trim(map(values, function(v) {
+          str_split(v, r"{\[}")[[1]][1]
+        }))
+      } # remove trailing source annotation (e.g. [Source:HGNC Symbol;Acc:HGNC:5])
+    ) %>%
+    mutate_at(
+      vars(canonical),
+      function(values) {
+        as_vector(
+          map(
+            str_trim(values),
+            function(v) {
+              if (is.na(v)) {
+                NA
+              } else if (v == "1") {
+                TRUE
+              } else if (v == "0") {
+                FALSE
+              }
+            }
+          )
         )
-
+      }
+    )
+  }else {
+  t2g <- biomaRt::getBM(
+  attributes = attributes,
+  mart = mart,
+  useCache = FALSE
+  )
+  t2g <- t2g %>%
+    rename(
+      target_id = ensembl_transcript_id,
+      ens_gene = ensembl_gene_id,
+      ext_gene = external_gene_name,
+      gene_desc = description,
+      canonical = transcript_is_canonical,
+    ) %>%
+    mutate_at(
+      vars(gene_desc),
+      function(values) {
+        str_trim(map(values, function(v) {
+          str_split(v, r"{\[}")[[1]][1]
+        }))
+      } # remove trailing source annotation (e.g. [Source:HGNC Symbol;Acc:HGNC:5])
+    ) %>%
+    mutate_at(
+      vars(canonical),
+      function(values) {
+        as_vector(
+          map(
+            str_trim(values),
+            function(v) {
+              if (is.na(v)) {
+                NA
+              } else if (v == "1") {
+                TRUE
+              } else if (v == "0") {
+                FALSE
+              }
+            }
+          )
+        )
+      }
+    )
+  }
```

Comment:
This code block looks quite redundant. Please avoid copy-pasting of code and use function calls instead. Further, please add comments explaining what you do and why you do it for each operation.

Reply:
Updated the code to reduce redundancy. Added comments explaining each operation.